extensions/GitExtension/FileExplorerGitIntegration/Models/GitExecute.cs

```diff
@@ -42,18 +42,25 @@ public static GitCommandRunnerResultInfo ExecuteGitCommand(string gitApplication
 
                 // Add timeout for 1 minute
                 process.WaitForExit(TimeSpan.FromMinutes(1));
+
+                if (process.ExitCode != 0)
+                {
+                    Log.Error("Execute Git process exited unsuccessfully with exit code {ExitCode}", process.ExitCode);
+                    return new GitCommandRunnerResultInfo(ProviderOperationStatus.Failure, output, "Execute Git process exited unsuccessfully", string.Empty, null, arguments, process.ExitCode);
+                }
+
                 return new GitCommandRunnerResultInfo(ProviderOperationStatus.Success, output);
             }
             else
             {
                 Log.Error("Failed to start the Git process: process is null");

```

Comment:
NIT: remove `process is null` The definition of a null process is one that did not start.